core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/CompressRequestStage.java

```diff
@@ -56,23 +60,27 @@ public SdkHttpFullRequest.Builder execute(SdkHttpFullRequest.Builder input, Requ
 
         Compressor compressor = resolveCompressionType(context.executionAttributes());
 
-        // non-streaming
-        if (!isStreaming(context)) {
-            ContentStreamProvider wrappedProvider = input.contentStreamProvider();
-            ContentStreamProvider compressedStreamProvider = () -> compressor.compress(wrappedProvider.newStream());
-            input.contentStreamProvider(compressedStreamProvider);
-            input = updateContentEncodingHeader(input, compressor);
-            return updateContentLengthHeader(input);
+        // non-streaming OR transfer-encoding chunked
+        if (!isStreaming(context) || isTransferEncodingChunked(input)) {
```

Comment:
In which case we use isTransferEncodingChunked for non streaming ? I was assuming that transferEncodingChunked is always for streaming.
How is isTransferEncodingChunked() set on the input  ? is it via user or on the model itself